lib/kennel/models/slo.rb

```diff
@@ -16,9 +17,15 @@ class Slo < Record
         tags: -> { @project.tags },
         query: -> { DEFAULTS.fetch(:query) },
         description: -> { DEFAULTS.fetch(:description) },
-        monitor_ids: -> { DEFAULTS.fetch(:monitor_ids) }
+        monitor_ids: -> { DEFAULTS.fetch(:monitor_ids) },
+        thresholds: -> { DEFAULTS.fetch(:thresholds) }
       )
 
+      def initialize(*)
+        super
+        raise ValidationError, "Threshold warning must be greater-than critical value" if thresholds.any? { |t| t[:warning].to_f <= t[:critical].to_f }
```

Comment:
that would fail if there is no warning though ? ... are warnings required ?

Reply:
Trying to find the docs for the SLO Widget api . . . do you have a link?  I can add it to the top of the SLO model file for future reference

Reply:
The UI allows the `Warning` to be empty.  I'll address this in another commit.

Reply:
yeah good idea ... can also verify via the UI if it lets you not enter a warning level